### DISCUSSION TITLE
Setups that are known to work

### DISCUSSION THREAD

**Discussion by PaulWolfram** [JGCRI/gcam-core#166] [Installing and running GCAM]
I love this new discussion forum! To start things off, I was wondering whether folks would be willing to share their setups that they use to run GCAM. As a new user, I had to try a bunch of different combinations using various software, library and R package versions in order to get things to work. I hope this will help new users to get things going faster. So here is my working setup that I use to run GCAM locally (I hope I am not missing anything):

- OS: Windows 10 x64 
- R 4.1.0 (packages: readr 1.4, dplyr 0.85; at least readr 2.0 caused problems)
- Rtools 4.0
- Visual Studio 2017 Community 15.0 (installed C++ and Windows 10 SDK workloads and most if not all individual components that mention "C++") 
- Windows 8.1 SDK (not sure I needed to install that)
- Java Development Kit 16.0.2 x64 
- BaseX 9.5.0
- Git Bash (Usually does not cause problems, I think any version will work) 
- Boost library 1.70.0 (1.76.0 caused problems)
- Xerces C 3.1 library copied from GCAM release version 5.2
- Jars library copied from GCAM release version 5.4
- TBB library 2021 2.0 Win ([direct download link](oneapi-tbb-2021.2.0-win.zip))
- Eigen library (version 3.4) ([direct download link](https://gitlab.com/libeigen/eigen/-/archive/master/eigen-master.tar.gz))

**Comment by crvernon** [comment]
👋 @PaulWolfram 

@rjplevin used to have a nice Python script that could be used to install GCAM on Windows, Mac, and Linux (see https://pygcam.readthedocs.io/en/master/install.html#installing-gcam-and-java).  I think this was used for an older version of GCAM though. 

@rjplevin have you by chance updated the `install-gcam.py` script lately to run with newer versions of GCAM?

**Comment by rjplevin** [comment]
Great idea to have a discussion forum!

No, I haven't updated the script, which I wrote in 2016 for GCAM v4.3.

**Reply by zebra-xs** [reply]
Dear PaulWolfram: From your post, I get the impression that, it doesn't seem easy to use GCAM, in fact we are energy analyst rather than an IT specialist, this model should be well prepared or almost ready for professional people to perform the system analysis, instead of asking them to do a lot of coding or programming work. So how to get it started in a good way from scrach ? thanks

**Reply by Yusheng63** [reply]
Surprised that no response for Zebra-XS's suggestion / query about 'start from scratch'. For a model as impactful as GCAM, more detailed instruction for a non-IT user is expected. I am a potential new user. I had downloaded version 7 but struggling to get it running. Any help will be appreciated.

**Comment by WineRosemob** [comment]
I'm having problems compiling GCAM from source, first of all, when setting environment variables, there is no way to look up the corresponding environment variables in the gcam-core-master file to find out where the corresponding links should be, for example, in config.system GCAMLIB_HOME =/pic/projects/GCAM/ GCAM-libraries, there is no way to find the file here, and there is no way to get the rest of the environment variables that use the softlink, so what should I do?

**Reply by WineRosemob** [reply]
If there is anyone here who would like to help me, I would appreciate it and can be contacted via email, thanks!

**Reply by WineRosemob** [reply]
I have already done, There are too many things to do for preparing this model
